spring-data-azure-documentdb/src/main/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplate.java

```diff
@@ -43,71 +46,145 @@ public DocumentDbTemplate(DocumentDbFactory documentDbFactory,
+                    .getResource();
+
+            if (resource instanceof Document) {
+                final Document originalDoc = (Document) resource;
+
+                documentDbFactory.getDocumentClient().replaceDocument(
+                        originalDoc.getSelfLink(),
+                        dbConverter.convertToDocument(object),
+                        null);
+            } else {
+                LOGGER.error("invalid Document to update {}", resource.getSelfLink());
+            }
+        } catch (DocumentClientException ex) {
+            throw new RuntimeException(ex.getMessage());
+        }
+    }
+
+    public <T> List<T> findAll(Class<T> entityClass) {
+        throw new UnsupportedOperationException("not supported");
+    }
+
+    public void deleteAll(String collectionName) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("execute deleteCollection in database {} collection {} with id {}",
+                    this.databaseName, collectionName);
+        }
+
+        try {
+            documentDbFactory.getDocumentClient()
+                    .deleteCollection(getCollectionLink(this.databaseName, collectionName), null);
+        } catch (DocumentClientException ex) {
+            if (ex.getStatusCode() == 404) {
+                LOGGER.warn("deleteAll in database {} collection {} met NOTFOUND: \n{}",
+                        this.databaseName, collectionName, ex.getMessage());
+            } else {
+                throw new RuntimeException(ex.getMessage());
+            }
+        }
+
+    }
+
+    public <T> List<T> findAll(String collectionName, final Class<T> entityClass) {
         final List<Document> results = documentDbFactory.getDocumentClient()
-                .queryDocuments(getCollectionLink(this.databaseName, getCollectionName(entityClass)), query, null)
+                .queryDocuments(getCollectionLink(this.databaseName, collectionName),
+                        "SELECT * FROM c", null)
                 .getQueryIterable().toList();
 
-        if (results == null || results.size() == 0) {
-            return null;
-        } else {
-            final Document d = results.get(0);
-            return dbConverter.convertFromDocument(d, entityClass);
+        final List<T> entities = new ArrayList<T>();
+
+        for (int i = 0; i < results.size(); i++) {
+            final T entity = dbConverter.convertFromDocument(results.get(i), entityClass);
+            entities.add(entity);
         }
+
+        return entities;
     }
 
-    public <T> void delete(T objectToRemove) {
-        throw new UnsupportedOperationException("not supported");
+    public String getCollectionName(Class<?> entityClass) {
+
+        return entityClass.getSimpleName();
     }
 
-    private Database getDb(String dbName) {
+    private Database createOrGetDatabase(String dbName) {
         final String query = "SELECT * FROM root r WHERE r.id='" + dbName + "'";
```

Comment:
I think below code section 

                if (resource instanceof Database) {
                    db = (Database) resource;
                }
                return db;

should be changed to 

                if (resource instanceof Database) {
                    return  (Database) resource;
                } else {
                     return null;
                }

Although origin code works, the logic is not correct.